src/python/pants/backend/python/packaging/pyoxidizer/rules_integration_test.py

```diff
@@ -56,19 +58,31 @@ def test_end_to_end() -> None:
         ),
     }
     with setup_tmpdir(sources) as tmpdir:
-        args = [
+        package_args = [
             "--backend-packages=['pants.backend.python', 'pants.backend.experimental.python.packaging.pyoxidizer']",
             f"--source-root-patterns=['/{tmpdir}']",
             "package",
             f"{tmpdir}/hellotest:bin",
         ]
-        result = run_pants(args)
-        result.assert_success()
+        package_result = run_pants(package_args)
+        package_result.assert_success()
 
         # Check that the binary is executable.
         bin_path = next(Path("dist", f"{tmpdir}.hellotest", "bin").glob("*/debug/install/bin"))
-        bin_stdout = subprocess.run([bin_path], check=True, stdout=subprocess.PIPE).stdout
-        assert bin_stdout == b"Hello world!\n"
+        bin_result = subprocess.run([bin_path], stdout=subprocess.PIPE)
+        assert bin_result.returncode == 42
+        assert bin_result.stdout == b"Hello world!\n"
+
+        # Performing `./pants run` check immediately after `./pants package`, in order to re-use package results and save time
```

Comment:
Could instead add to the test's docstring this:

```
* `run` vs `package`
```

The docstring already explains we are bundling things together for the sake of performance